packages/undo-manager/src/index.ts

```diff
@@ -3,24 +3,85 @@
  */
 import isShallowEqual from '@wordpress/is-shallow-equal';
 
-/** @typedef {import('./types').HistoryRecord}  HistoryRecord */
-/** @typedef {import('./types').HistoryChange}  HistoryChange */
-/** @typedef {import('./types').HistoryChanges} HistoryChanges */
-/** @typedef {import('./types').UndoManager} UndoManager */
+/**
+ * Represents a single change in history.
+ */
+export interface HistoryChange< T = unknown > {
+	/** The previous value */
+	from: T;
+	/** The new value */
+	to: T;
+}
+
+/**
+ * Represents changes for a single item.
+ */
+export interface HistoryChanges< T = unknown > {
+	/** The identifier for the item being changed */
+	id: string | Record< string, unknown >;
+	/** The changes made to the item */
+	changes: Record< string, HistoryChange< T > >;
+}
+
+/**
+ * Represents a record of history changes.
+ */
+export type HistoryRecord< T = unknown > = HistoryChanges< T >[];
+
+/**
+ * The undo manager interface.
+ */
+export interface UndoManager< T = unknown > {
```

Comment:
We already have all of these interfaces in the `types.ts` file, can we please update those and use them in this file?